client/state/data-layer/wpcom/sites/media/test/index.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { noop, times } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import useNock from 'test/helpers/use-nock';
+import { useSandbox } from 'test/helpers/use-sinon';
+import {
+	MEDIA_FILE_UPLOAD,
+	MEDIA_FILE_UPLOAD_FAILURE,
+	MEDIA_FILE_UPLOAD_SUCCESS,
+	MEDIA_ITEMS_RECEIVE
+} from 'state/action-types';
+import { uploadFile, enqueueFileUpload } from 'state/media/actions';
+import { requestFileUpload, uploadNext, maybeUploadFirst } from '../';
+
+describe( 'handler', () => {
+	let state;
+	const store = { dispatch: noop, getState: noop };
+
+	useSandbox( ( sandbox ) => {
+		sandbox.stub( store, 'dispatch' )
+			// When called with thunk, call on thunk with dispatch
+			.withArgs( sinon.match.func ).callsArgWith( 0, store.dispatch );
+		sandbox.stub( store, 'getState', () => state );
+	} );
+
+	function stubStateUploadsInProgress( uploads = 0 ) {
+		state = {
+			media: {
+				uploadQueue: times( uploads, () => [ 2916284, 'https://wordpress.com/i/stats-icon.gif' ] )
+			}
+		};
```

Comment:
This kind of state mocking is why I have tried to start writing composed selectors. On the negative side, we get more function call overhead, but on the positive side, we get easy mocking and more composable state code.

For an example see [the AT-AT selectors](https://github.com/Automattic/wp-calypso/blob/f69d6577bb6c5232d993365b4d440f42c8b6a254/client/state/automated-transfer/selectors.js#L47)